cpp/arcticdb/pipeline/read_frame.cpp

```diff
@@ -1019,12 +1019,12 @@ folly::Future<std::vector<VariantKey>> fetch_data(
     {
```

Comment:
I still think that we should allocate continuations in the heap, instead of the stack, to avoid problems like this one https://github.com/man-group/ArcticDB/actions/runs/5136691407/jobs/9245918336?pr=435.

If an object is created on the stack within a function, then it will be automatically destroyed when the function execution completes, regardless if it has been moved to another thread that is still executing or not. 

Since we are using folly futures, it's possible that some threads running continuations will still be executing when we terminate this function. These continuations may access captured members, including shared pointers that have been destroyed because no one is pointing to the object anymore.

I would personally use a make_shared here for the continuations.

Reply:
The continuations are being moved into bach_read_compressed, there is no need for a shared_ptr that I can see

Reply:
The only thing that was being captured by reference (and which could therefore go out of scope) is the buffers, which is already a shared_ptr, which is why I'm now passing it by value

Reply:
yes I see your point, and if the members from continuations didn't contain any shared pointer we would still be fine...However, considering my last paragraph, there is a rare case where the shared pointers inside the captured per copy members may reach a reference count of zero, resulting in the destruction of the object they point to.

Reply:
yes, correct. As I commented, in this case we are fine